src/CommunityToolkit.Maui/Extensions/AppThemeExtension.shared.cs

```diff
@@ -0,0 +1,91 @@
+﻿namespace CommunityToolkit.Maui.Extensions;
+
+/// <summary>
+/// A XAML markup extension that enables using <see cref="AppThemeColor"/> and <see cref="AppThemeResource"/> from XAML.
+/// </summary>
+[ContentProperty(nameof(Key))]
+public sealed class AppThemeExtension : IMarkupExtension<BindingBase>
```

Comment:
We probably want to rename this because now the extension is `AppTheme` which collides with the `AppTheme` in .NET MAUI and Visual Studio suggests the wrong namespace to be imported. Unless you use the full name `AppThemeExtension` but not sure if that's desirable.